### PR TITLE
NEW resource address

### PR DESCRIPTION
**Pull request by W1W1-M**
# NEW resource address
As proposed in #26230 this adds full address to a resource

### CODE REVIEW

File: htdocs/resource/class/dolresource.class.php [hunk 1]
```diff
@@ -98,69 +110,65 @@ class Dolresource extends CommonObject
...
-	 *  @param	DoliDb		$db      Database handler
+	 *  @param DoliDb $db      Database handler
 	 */
-	public function __construct($db)
+	public function __construct(DoliDb $db)
 	{
 		$this->db = $db;
+		$this->element = 'dolresource';
+		$this->table_element = 'resource';
+		$this->picto = 'resource';
```
Comment (eldy) [inline htdocs/resource/class/dolresource.class.php:120]: Why moving the value of final properties into the constructor ? They are properties that should not be modified, so no reason to need to instantiate an instance to get their value.

Comment (W1W1-M) [reply htdocs/resource/class/dolresource.class.php:120]: I moved the properties to constructor to centralize variable initialization and make good use of the class constructor, just a coding preference. 

However if these properties should not be modified and are to be accessed without instantiating maybe the better solution is to declare them clearly as class constants ?

Comment (W1W1-M) [reply htdocs/resource/class/dolresource.class.php:120]: But then I suppose they would have to be redefined as constants for CommonObject and all other child classes ...

Comment (eldy) [reply htdocs/resource/class/dolresource.class.php]: Yes, they should have been set as const. But this was coded a long time ago and changing property to const now is too complex and not a priority. 
In a future we may use the declaration with final https://www.php.net/manual/fr/language.oop5.final.php
public final string $element = 'abcdef';
but we need to wai min php 8.1 for that. For the moment, we should keep them in declaration.

Comment (W1W1-M) [reply htdocs/resource/class/dolresource.class.php]: Thanks

**Pull request merged**